h/services/annotation_stats.py

```diff
@@ -16,36 +14,27 @@ class AnnotationStatsService(object):
     def __init__(self, request):
         self.request = request
 
-    def user_annotation_counts(self, userid):
-        """Return the count of annotations for this user."""
-
-        annotations = self.request.db.query(Annotation). \
-            filter_by(userid=userid, deleted=False). \
-            options(sa.orm.load_only('groupid', 'shared')).subquery()
-        grouping = sa.case([
-            (sa.not_(annotations.c.shared), 'private'),
-            (annotations.c.groupid == '__world__', 'public'),
-        ], else_='group')
-
-        result = dict(self.request.db.query(grouping, sa.func.count(annotations.c.id)).group_by(grouping).all())
```

Comment:
Do you know what the generated SQL looked like for this query. The initial `annotations` query doesn't look inefficient (`SELECT groupid, shared FROM annotation WHERE userid := userid, deleted = false`), but I can't immediately see how the rest of the SQLAlchemy query translates to an actual DB query.

Reply:
SELECT CASE WHEN NOT anon_2.shared THEN %(param_1)s WHEN (anon_2.groupid = %(groupid_1)s) THEN %(param_2)s ELSE %(param_3)s END AS anon_1, count(anon_2.id) AS count_1 FROM (SELECT annotation.id AS id, annotation.groupid AS groupid, annotation.shared AS shared FROM annotation WHERE annotation.deleted = ? AND annotation.userid = %(userid_1)s) AS anon_2 GROUP BY CASE WHEN NOT anon_2.shared THEN %(param_1)s WHEN (anon_2.groupid = %(groupid_1)s) THEN %(param_2)s ELSE %(param_3)s END